src/Symfony/Component/Notifier/Channel/BrowserChannel.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Channel;
+
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Notifier\Notification\Notification;
+use Symfony\Component\Notifier\Receiver\Receiver;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @experimental in 5.0
+ */
+final class BrowserChannel implements ChannelInterface
+{
+    private $stack;
+
+    public function __construct(RequestStack $stack)
+    {
+        $this->stack = $stack;
+    }
+
+    public function notify(Notification $notification, Receiver $receiver, string $transport = null, MessageBusInterface $bus = null): void
+    {
+        if (null === $request = $this->stack->getCurrentRequest()) {
+            return;
+        }
+
+        $message = $notification->getSubject();
+        if ($notification->getEmoji()) {
+            $message = $notification->getEmoji().' '.$message;
+        }
+        $request->getSession()->getFlashBag()->add('notice', $message);
```

Comment:
Why did you choose `notice`? I would use `notifier` if something should be hardcoded

Reply:
Using `notification` is indeed a good idea, changed.